src/protocol/mod.rs

```diff
@@ -1023,6 +1037,51 @@ mod load {
         }
     }
 
+    mod regex_matching {
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        #[test]
+        fn parses_a_regex_command_matcher() -> R<()> {
+            let step = test_parse_one(
+                r"
+                    |protocols:
+                    |  - protocol:
+                    |      - regex: \d
+                ",
+            )?
+            .steps[0]
+                .clone();
+            match step.command {
+                CommandMatcher::RegexMatch(regex) => assert_eq!(regex.as_str(), "^\\d$"),
```

Comment:
i'm not in love with the fact that i had to inject the anchors in here, but i was trying to avoid using `CommandMatcher::regex_match` to build this based on our conversation today.

Reply:
Yeah, I wouldn't use `regex_match`. If we make the change to `RegexMatch(String)` then we would leave out the anchors of the `String`. That way our in-memory representation is very close to the given yaml, which I kind of like.